src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,188 @@
+            if (this.secondWord() == null) {
+                throw new DukeException("Please add the task name");
+            } else {
+                return Commands.of(cmd, this.secondWord());
+            }
+        }
+
+        if (cmd == Commands.COMMANDS.BY || cmd == Commands.COMMANDS.FROM || cmd == Commands.COMMANDS.TO) {
+            String restOfCommand = this.secondWord().trim();
+            try {
+                LocalDateTime dateTime = LocalDateTime.parse(restOfCommand, Duke.FORMAT);
+                return Commands.of(cmd, dateTime);
+            } catch (DateTimeParseException e) {
+                throw new DukeDateTimeParseException(cmd
+                        + ": The format for dates&time is 'dd-MM-yyyy hhmm'");
+            }
+        }
+
+        if (cmd == Commands.COMMANDS.MARK || cmd == Commands.COMMANDS.UNMARK || cmd == Commands.COMMANDS.DELETE) {
+            try {
+                int index = Integer.parseInt(this.secondWord());
+                return Commands.of(cmd, index);
+            } catch (NumberFormatException | NullPointerException e) {
+                throw new DukeNumberFormatException("Place a number after the command");
+            }
+        }
+
+        if (cmd == Commands.COMMANDS.DEADLINE) {
+            try {
+                String task = this.phaseParse();
+                String command2 = this.phaseTwo();
+                if (task != null) {
+                    Parser phaseTwo = new Parser(command2);
+                    Commands c = phaseTwo.parse();
+                    if (c.checkCommand(Commands.COMMANDS.BY)) {
+                        return Commands.of(Commands.COMMANDS.DEADLINE, task, c);
+                    }
+                } else {
+                    throw new DukeException("Please add the task name");
+                }
+            } catch (DukeUnknownCommandException e) {
+                throw new DukeNullPointerException("The format for the command is: deadline task /by date&time");
+            } catch (NullPointerException e) {
+                throw new DukeNullPointerException("The format for the command is: deadline task /by date&time");
+            }
+        }
+
+        if (cmd == Commands.COMMANDS.EVENT) {
+            try {
+                String task = this.phaseParse();
+                String command2 = this.phaseTwo();
+                String command3 = this.phaseThree();
+                if (task != null) {
+                    Parser phaseTwo = new Parser(command2);
+                    Commands c1 = phaseTwo.parse();
+                    Parser phaseThree = new Parser(command3);
+                    Commands c2 = phaseThree.parse();
+                    if (c1.checkCommand(Commands.COMMANDS.FROM) && c2.checkCommand(Commands.COMMANDS.TO)) {
+                        if (c1.compareTime(c2)) {
+                            return Commands.of(Commands.COMMANDS.EVENT, task, c1, c2);

```

Comment:
Quite deep code indentation here, is there a way to simplify this?

Reply:
I'll try, but what do you suggest.